go-controller/pkg/ovn/egressip.go

```diff
@@ -1120,25 +1126,40 @@ func (oc *Controller) syncEgressIPs(eIPs []interface{}) {
 
 // This function implements a portion of syncEgressIPs.
 // It removes OVN logical router policies used by EgressIPs deleted while ovnkube-master was down.
+// It also removes stale nexthops from router policies used by EgressIPs.
 // Upon failure, it may be invoked multiple times in order to avoid a pod restart.
-func (oc *Controller) syncStaleEgressReroutePolicy(egressIPToPodIPCache map[string]sets.String) error {
+func (oc *Controller) syncStaleEgressReroutePolicy(egressIPCache map[string]egressIPCacheEntry) error {
 	logicalRouter := nbdb.LogicalRouter{}
 	logicalRouterPolicyRes := []nbdb.LogicalRouterPolicy{}
+	logicalRouterPolicyStaleNexthops := make(map[string]nbdb.LogicalRouterPolicy)
 	opModels := []libovsdbops.OperationModel{
 		{
 			ModelPredicate: func(lrp *nbdb.LogicalRouterPolicy) bool {
 				if lrp.Priority != types.EgressIPReroutePriority {
 					return false
 				}
 				egressIPName := lrp.ExternalIDs["name"]
-				podIPCache, exists := egressIPToPodIPCache[egressIPName]
+				cacheEntry, exists := egressIPCache[egressIPName]
 				splitMatch := strings.Split(lrp.Match, " ")
 				logicalIP := splitMatch[len(splitMatch)-1]
 				parsedLogicalIP := net.ParseIP(logicalIP)
-				if !exists || !podIPCache.Has(parsedLogicalIP.String()) {
-					klog.Infof("syncStaleEgressReroutePolicy will delete %s: %v", egressIPName, lrp)
+				if !exists || !cacheEntry.podIPs.Has(parsedLogicalIP.String()) {
+					klog.Infof("syncStaleEgressReroutePolicy will delete %s due to stale logical ip: %v", egressIPName, lrp)
 					return true
 				}
+				// Check for stale nexthops that may exist in the logical router policy and store that in logicalRouterPolicyStaleNexthops
+				staleNextHops := sets.NewString()
+				for _, nexthop := range lrp.Nexthops {
+					if !cacheEntry.gatewayRouterIPs.Has(nexthop) {
+						staleNextHops.Insert(nexthop)
+					}
+				}
+				if staleNextHops.Len() > 0 {
```

Comment:
what happens if this lrp has 0 valid nexthops. In other words, you found LRP X with 1 nexthop, it was stale, and now you remove the next hop so the LRP now has 0 next hops. Shouldn't we just delete the entire LRP if that is the case?

Reply:
hm.... good point. Let me add the logic there to remove the row if there are no valid nextHops left.

Reply:
Update: I made the changes, but after looking back at the code, I am now thinking it is not really necessary.
An egress ip row would have been deleted when there are no nexthops (nodes) so this additional logic may not
buy us anything. So... @trozet if you strongly feel that is a good thing to have, lmk and I will update the PR with it:
Ref: https://gist.github.com/flavio-fernandes/751091b6a75dad0b68493813c5c1dac0

Reply:
After talking some more with @trozet , I found a simpler way of accomplishing the same outcome:
```
if !exists || cacheEntry.gatewayRouterIPs.Len() == 0 || !cacheEntry.podIPs.Has(parsedLogicalIP.String()) {
	klog.Infof("syncStaleEgressReroutePolicy will delete %s due to no nexthop or stale logical ip: %v", egressIPName, lrp)
```
PR updated.